Various Hedera go sdk issues/inconsistencies found while working on wallet connect support

### Description

While working on a parser I need for all 36 transaction types to support wallet connect, I found a number of issues or inconsistencies. Rather than explain them all here, I have simply submitted unit tests in the attached files. It's possible you may disagree on whether or not a couple of these are defects.

The most important one that I need fixed for the Wallawallet wallet connect release is the one named TestHGoSdkIssue_Incorrectly_Throws_Unsupported_Transaction_Type_Error, where I cannot reconstitute an AccountUpdateTransaction from the transaction bytes. I have not yet attempted to debug this further than my own code.

[HederaSdkDefects_test.go.txt](https://github.com/hashgraph/hedera-sdk-go/files/12689544/HederaSdkDefects_test.go.txt)
[test-util.go.txt](https://github.com/hashgraph/hedera-sdk-go/files/12689545/test-util.go.txt)

I've included relevant portions of the output of running the tests below so you can see at a glance what the issues are:

```
Contents Base64:  
--- FAIL: TestHGoSdkIssue_FileAppendTransaction_Missing_Contents_After_Reconstitution (0.12s)
    HederaSdkDefects_test.go:86: error parsing transaction bytes: Expected to find Contents for FileAppendTransaction, but there were none.
Message Base64:  dGVzdCB0b3BpYyBtZXNzYWdl
Message Base64:  
--- FAIL: TestHGoSdkIssue_TopicMessageSubmitTransaction_Missing_Message_After_Reconstitution (0.12s)
    HederaSdkDefects_test.go:107: error parsing transaction bytes: Expected to find Contents for FileAppendTransaction, but there were none.
Assessment Method:  FeeAssessmentMethodInclusive
--- FAIL: TestHGoSdkIssue_CustomFractionFee_Missing_SetAssessmentMethod_Method (0.00s)
    HederaSdkDefects_test.go:116: There is currently no SetAssessmentMethod() for CustomFractionalFee in golang, but it exists in JS SDK and as far as I can tell is needed
--- FAIL: TestHGoSdkIssue_SystemDeleteTransaction_Type_Inconsistent_For_GetExpirationTime (0.09s)
    HederaSdkDefects_test.go:135: expected GetExpirationTime() to return a time.Time type (as that's what the setter uses), not: int64
--- FAIL: TestHGoSdkIssue_ContractCreateTransaction_Uses_Inconsistent_Type_For_MaxAutomaticTokenAssociations (0.10s)
    HederaSdkDefects_test.go:183: expected GetMaxAutomaticTokenAssociations() for AccountCreateTransaction and ContractCreateTransaction to return value of same type, but instead they were uint32 and int32
--- FAIL: TestHGoSdkIssue_AutoRenewAccount_Not_Correctly_Set_For_TokenCreateTransaction (0.10s)
    HederaSdkDefects_test.go:229: expected autoRenewAccount to be the one that was explictly set, which is 0.0.122837, but got 0.0.2240404
--- FAIL: TestHGoSdkIssue_TopicCreateTransaction_May_Be_Missing_expirationTime_Property (0.00s)
    HederaSdkDefects_test.go:251: The TopicCreateTransaction is missing the expirationTime property, yet the TopicUpdateTransaction has that property. This might be a bug.
FAIL
exit status 1

``` 


### Steps to reproduce

1. Rename the files to .go files
2. My project has just these dependencies: require (
	github.com/hashgraph/hedera-sdk-go/v2 v2.29.0
	github.com/stretchr/testify v1.8.4
	google.golang.org/protobuf v1.31.0
)
3. Once configured, just run "go test", or "go test -v -run <testname>" for an individual test.

### Additional context

_No response_

### Hedera network

testnet

### Version

v2.29.0

### Operating system

macOS

And another serialization issue for TopicMemo for TopicCreateTransaction:

=== RUN   TestHGoSdkIssue_TopicCreateTransaction_Missing_TopicMemo_After_Reconstitution
b64Transaction: | CrYBKrMBCq4BChUKDAju2suoBhCt56O/AhIFGJTfiAESAhgEGICg2eYdIgMItAEyHXRlc3QgdG9waWMgdXBkYXRlIHRyYW5zYWN0aW9uwgFmCg90ZXN0IHRvcGljIG1lbW8SIhIgXYnw0+3YHV7hy4qouJFxw2YVddamRpqKM5qWDTALN5waIhIgXYnw0+3YHV7hy4qouJFxw2YVddamRpqKM5qWDTALN5wyBQiAnJw5OgQY1b8HEgAKtgEqswEKrgEKFQoMCO7ay6gGEK3no78CEgUYlN+IARICGAcYgKDZ5h0iAwi0ATIddGVzdCB0b3BpYyB1cGRhdGUgdHJhbnNhY3Rpb27CAWYKD3Rlc3QgdG9waWMgbWVtbxIiEiBdifDT7dgdXuHLiqi4kXHDZhV11qZGmoozmpYNMAs3nBoiEiBdifDT7dgdXuHLiqi4kXHDZhV11qZGmoozmpYNMAs3nDIFCICcnDk6BBjVvwcSAAq2ASqzAQquAQoVCgwI7trLqAYQreejvwISBRiU34gBEgIYAxiAoNnmHSIDCLQBMh10ZXN0IHRvcGljIHVwZGF0ZSB0cmFuc2FjdGlvbsIBZgoPdGVzdCB0b3BpYyBtZW1vEiISIF2J8NPt2B1e4cuKqLiRccNmFXXWpkaaijOalg0wCzecGiISIF2J8NPt2B1e4cuKqLiRccNmFXXWpkaaijOalg0wCzecMgUIgJycOToEGNW/BxIA |
    HederaSdkDefects_test.go:256: error parsing transaction bytes: Expected topic memo to be "test topic memo" but was ""
--- FAIL: TestHGoSdkIssue_TopicCreateTransaction_Missing_TopicMemo_After_Reconstitution (0.10s)

--- a/file_append_transaction.go
+++ b/file_append_transaction.go
@@ -58,7 +58,7 @@ func _FileAppendTransactionFromProtobuf(transaction Transaction, pb *services.Tr
 	return &FileAppendTransaction{
 		Transaction: transaction,
 		maxChunks:   20,
-		contents:    make([]byte, 0),
+		contents:    pb.GetFileAppend().GetContents(),
 		chunkSize:   2048,
 		fileID:      _FileIDFromProtobuf(pb.GetFileAppend().GetFileID()),
 	}

--- a/file_append_transaction_unit_test.go
+++ b/file_append_transaction_unit_test.go
@@ -363,3 +363,32 @@ func TestUnitFileAppendTransactionCoverage(t *testing.T) {
 		b.AddSignature(newKey.PublicKey(), sig)
 	}
 }
+
+func TestUnitFileAppendTransactionSerialization(t *testing.T) {
+	t.Parallel()
+
+	fileID := FileID{File: 7}
+	nodeAccountID := []AccountID{{Account: 10}}
+	transactionID := TransactionIDGenerate(AccountID{Account: 324})
+
+	transaction, err := NewFileAppendTransaction().
+		SetTransactionID(transactionID).
+		SetNodeAccountIDs(nodeAccountID).
+		SetFileID(fileID).
+		SetContents([]byte("Hello, World")).
+		Freeze()
+	require.NoError(t, err)
+
+	txBytes, err := transaction.ToBytes()
+	require.NoError(t, err)
+
+	txParsed, err := TransactionFromBytes(txBytes)
+	require.NoError(t, err)
+
+	result, ok := txParsed.(FileAppendTransaction)
+	require.True(t, ok)
+
+	require.Equal(t, transactionID.AccountID, result.GetTransactionID().AccountID)
+	require.Equal(t, fileID, result.GetFileID())
+	require.Equal(t, transaction.GetContents(), result.GetContents())
+}

--- a/token_create_transaction.go
+++ b/token_create_transaction.go
@@ -774,7 +774,7 @@ func (transaction *TokenCreateTransaction) Freeze() (*TokenCreateTransaction, er
 }
 
 func (transaction *TokenCreateTransaction) FreezeWith(client *Client) (*TokenCreateTransaction, error) {
-	if transaction.autoRenewPeriod != nil && client != nil && !client.GetOperatorAccountID()._IsZero() {
+	if transaction.autoRenewAccountID == nil && transaction.autoRenewPeriod != nil && client != nil && !client.GetOperatorAccountID()._IsZero() {
 		transaction.SetAutoRenewAccount(client.GetOperatorAccountID())
 	}
 

--- a/token_create_transaction_unit_test.go
+++ b/token_create_transaction_unit_test.go
@@ -330,3 +330,14 @@ func TestUnitTokenCreateTransactionMock(t *testing.T) {
 	_, err = freez.Sign(newKey).Execute(client)
 	require.NoError(t, err)
 }
+
+func TestUnitTokenCreateAutoRenewAccountSetting(t *testing.T) {
+	client, err := _NewMockClient()
+	autoRenewAccount := AccountID{Account: 3}
+	frozenTx, err := NewTokenCreateTransaction().
+		SetAutoRenewAccount(autoRenewAccount).
+		SetAutoRenewPeriod(24 * time.Hour).
+		FreezeWith(client)
+	require.NoError(t, err)
+	require.Equal(t, autoRenewAccount, frozenTx.GetAutoRenewAccount())
+}

--- a/topic_create_transaction.go
+++ b/topic_create_transaction.go
@@ -66,7 +66,7 @@ func _TopicCreateTransactionFromProtobuf(transaction Transaction, pb *services.T
 		autoRenewAccountID: _AccountIDFromProtobuf(pb.GetConsensusCreateTopic().GetAutoRenewAccount()),
 		adminKey:           adminKey,
 		submitKey:          submitKey,
-		memo:               pb.GetContractCreateInstance().GetMemo(),
+		memo:               pb.GetConsensusCreateTopic().GetMemo(),
 		autoRenewPeriod:    &autoRenew,
 	}
 }

--- a/topic_create_transaction_unit_test.go
+++ b/topic_create_transaction_unit_test.go
@@ -302,3 +302,38 @@ func TestUnitTopicCreateTransactionMock(t *testing.T) {
 	_, err = freez.Sign(newKey).Execute(client)
 	require.NoError(t, err)
 }
+
+func TestUnitTopicCreateTransactionSerialization(t *testing.T) {
+	t.Parallel()
+
+	nodeAccountID := []AccountID{{Account: 10}}
+	transactionID := TransactionIDGenerate(AccountID{Account: 324})
+	newKey, err := PrivateKeyGenerateEd25519()
+	require.NoError(t, err)
+
+	topicCreate, err := NewTopicCreateTransaction().
+		SetTransactionID(transactionID).
+		SetAdminKey(newKey).
+		SetNodeAccountIDs(nodeAccountID).
+		SetSubmitKey(newKey).
+		SetTopicMemo("ad").
+		SetAutoRenewPeriod(time.Second * 30).
+		Freeze()
+	require.NoError(t, err)
+
+	transactionBytes, err := topicCreate.ToBytes()
+	require.NoError(t, err)
+
+	txParsed, err := TransactionFromBytes(transactionBytes)
+	require.NoError(t, err)
+
+	result, ok := txParsed.(TopicCreateTransaction)
+	require.True(t, ok)
+
+	require.Equal(t, topicCreate.GetTopicMemo(), result.GetTopicMemo())
+	require.Equal(t, topicCreate.GetAutoRenewPeriod(), result.GetAutoRenewPeriod())
+	adminKey, _ := result.GetAdminKey()
+	require.Equal(t, newKey.PublicKey(), adminKey)
+	submitKey, _ := result.GetSubmitKey()
+	require.Equal(t, newKey.PublicKey(), submitKey)
+}

--- a/topic_message_submit_transaction.go
+++ b/topic_message_submit_transaction.go
@@ -58,7 +58,7 @@ func _TopicMessageSubmitTransactionFromProtobuf(transaction Transaction, pb *ser
 	tx := &TopicMessageSubmitTransaction{
 		Transaction: transaction,
 		maxChunks:   20,
-		message:     make([]byte, 0),
+		message:     pb.GetConsensusSubmitMessage().GetMessage(),
 		topicID:     _TopicIDFromProtobuf(pb.GetConsensusSubmitMessage().GetTopicID()),
 	}
 

--- a/topic_message_submit_transaction_unit_test.go
+++ b/topic_message_submit_transaction_unit_test.go
@@ -291,3 +291,34 @@ func TestUnitTopicMessageSubmitTransactionCoverage(t *testing.T) {
 		b.AddSignature(newKey.PublicKey(), sig)
 	}
 }
+
+func TestUnitTopicMessageSubmitTransactionSerialization(t *testing.T) {
+	t.Parallel()
+
+	topic := TopicID{Topic: 3}
+	nodeAccountID := []AccountID{{Account: 10}}
+	transactionID := TransactionIDGenerate(AccountID{Account: 324})
+
+	transaction, err := NewTopicMessageSubmitTransaction().
+		SetTransactionID(transactionID).
+		SetNodeAccountIDs(nodeAccountID).
+		SetTopicID(topic).
+		SetMessage([]byte("nothing to see here")).
+		SetMaxChunks(30).
+		SetTransactionMemo("no").
+		Freeze()
+	require.NoError(t, err)
+
+	txBytes, err := transaction.ToBytes()
+	require.NoError(t, err)
+
+	txParsed, err := TransactionFromBytes(txBytes)
+	require.NoError(t, err)
+
+	result, ok := txParsed.(TopicMessageSubmitTransaction)
+	require.True(t, ok)
+
+	require.Equal(t, transactionID.AccountID, result.GetTransactionID().AccountID)
+	require.Equal(t, transaction.GetMessage(), result.GetMessage())
+	require.Equal(t, transaction.GetTransactionMemo(), result.GetTransactionMemo())
+}

